.github/workflows/kfp-backend-postgres-tests.yml

```diff
@@ -0,0 +1,245 @@
+name: KFP Backend Postgres Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      test_repository:
+        description: "owner/repo to checkout (empty = upstream)"
+        required: false
+        default: ""
+      test_ref:
+        description: "Git ref (branch/tag/SHA) to test"
+        required: true
+        default: "master"
+      test_paths:
+        description: "Go package paths"
+        required: true
+        default: "./backend/test/initialization"
+
+jobs:
+  postgres-pgx:
+    name: Backend Postgres (pgx via Kustomize+KinD)
+    runs-on: ubuntu-latest
+    continue-on-error: true # Temporarily allow failures; switch to false after LargeText/storage changes land
+
+    steps:
+      - name: Checkout target code
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.inputs.test_repository || github.repository }}
+          ref: ${{ github.event.inputs.test_ref }}
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+          cache: true
+
+      - name: Create KinD cluster
+        uses: helm/kind-action@v1
+        with:
+          cluster_name: kfp-ci
+          wait: 300s
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@v4
+        with:
+          version: "latest"
+
+      - name: Install kustomize
+        uses: imranismail/setup-kustomize@v2
+        with:
+          kustomize-version: "5.4.2"
+
+      - name: Install utils (psql, nc, jq)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql-client netcat-openbsd jq
+
+      - name: Create namespace
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: kubectl create namespace "$NAMESPACE" || true
+
+      - name: Install default StorageClass for KinD (local-path)
```

Comment:
we have a `kind` cluster action at `.github/actions/kfp-cluster` which will use the kind action, can we just re-use that here instead of setting up the kind cluster from scratch?